protocol/contracts/templegold/DaiGoldAuction.sol

```diff
@@ -0,0 +1,262 @@
+
+/** 
+ * @title AuctionEscrow
+ * @notice Bidding token is deposited into this contract to bid on a share of distributed Temple Gold for an epoch.
+ *         Temple Gold acquired in past epochs can always be claimed. Once bidding for an epoch has ended, users cannot
+ *         claim their bid token and can claim their share of Temple Gold for epoch.
+ *         Elevated access can change bidding token for future epochs.
+ */
+contract DaiGoldAuction is IDaiGoldAuction, AuctionBase, TempleElevatedAccess {
+    using SafeERC20 for ITempleGold;
+    using SafeERC20 for IERC20;
+    using TempleMath for uint256;
+    
+    /// @notice Temple GOLD address
+    ITempleGold public immutable override templeGold;
+    /// @notice Token to bid for temple GOLD
+    IERC20 public override bidToken;
+    /// @notice Destination address for proceeds of fire ritual
+    address public immutable override treasury;
+    /// @notice Address that can trigger start of auction. address(0) means anyone
+    address public override auctionStarter;
+
+    /// @notice Keep track of next epoch auction Temple Gold amount
+    uint256 public override nextAuctionGoldAmount;
+    /// @notice last time rewards was notified
+    uint96 public override lastRewardNotificationTimestamp;
+
+    /// @notice Auction duration
+    uint64 public constant AUCTION_DURATION = 1 weeks;
+
+    AuctionConfig private auctionConfig;
+
+    constructor(
+        address _templeGold,
+        address _bidToken,
+        address _treasury,
+        address _rescuer,
+        address _executor
+    ) TempleElevatedAccess(_rescuer, _executor) {
+        templeGold = ITempleGold(_templeGold);
+        bidToken = IERC20(_bidToken);
+        treasury = _treasury;
+    }
+
+    /**
+     * @notice Set auction configuration
+     * @param _config Auction configuration
+     */
+    function setAuctionConfig(AuctionConfig calldata _config) external override onlyElevatedAccess {
+        if (_config.auctionStartCooldown == 0
+                || _config.auctionMinimumDistributedGold == 0
+                || _config.auctionsTimeDiff == 0) 
+            { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        auctionConfig = _config;
+
+        emit AuctionConfigSet(_currentEpochId, _config);
+    }
+
+    /**
+     * @notice Set address to trigger auction start. Zero address accepted
+     * @param _starter Auction starter
+     */
+    function setAuctionStarter(address _starter) external override onlyElevatedAccess {
+        /// @notice No zero address checks. Zero address is a valid input
+        auctionStarter = _starter; 
+        emit AuctionStarterSet(_starter);
+    }
+
+    /**
+     * @notice Set token used for bidding
+     * @param _bidToken Bid token
+     */
+    function setBidToken(address _bidToken) external override onlyElevatedAccess {
+        if (_bidToken == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
```

Comment:
No need to check if currentEpoch is in progress?
In other set functions including `setAuctionStarter` and `setAuctionConfig`

Of course, this functions can be called by only elevated accessed users, but maybe time miscalculation would happen?

Reply:
good call